saleor/core/prices.py

```diff
@@ -1,16 +1,18 @@
 from decimal import Decimal
-from typing import TYPE_CHECKING, Iterable, Union
+from typing import TYPE_CHECKING, Iterable, TypeVar
 
 from babel.numbers import get_currency_precision
 
 if TYPE_CHECKING:
     from django.db.models import Model
-    from prices import Money, TaxedMoney, TaxedMoneyRange
+    from prices import Money, TaxedMoney, TaxedMoneyRange  # noqa
 
-MoneyTypes = Union["TaxedMoney", "Money", Decimal, "TaxedMoneyRange"]
+MoneyType = TypeVar(
+    "MoneyType", "TaxedMoney", "Money", Decimal, "TaxedMoneyRange"  # noqa
```

Comment:
Why do you include new type as a `type` that belongs to?
`MoneyType = TypeVar("MoneyType")`

Reply:
I've had problems with mypy using `quantize_price` function. When passing `Decimal`, inferred return value was of type `TaxedMoney`. The return type should be exactly the same as parameter type and `TypeVar` is better suited for that case.

Reply:
Maybe we should better name that `MoneyType`.

Reply:
How about `PriceType`?